src/commands/logs.ts

```diff
@@ -13,15 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import queryString from 'query-string';
+
+import queryString, { parse } from 'query-string';
 import WebSocket from 'ws';
 
+import {
+  defaultLogColorizer,
+  noColorLogColorizer
+} from './logging/coloring';
+
+import {
+  jsonLogger,
+  keyvalueLogger,
+  Logger,
+  simpleLogger
+} from './logging/logger';
+
 import { controller } from '../api';
 import { Arguments, createCommand } from '../util';
 
-function handleLogData(logData: any): void {
-  const { msg: message } = logData;
-  console.log(message);
+function parseLogStream(line: string): any {
+  const logData = JSON.parse(line);
+  if (logData.structured !== undefined) {
+    return logData;
+  }
+
+  // FIXME: remove the following when structured log is supported in piper
+  const { msg } = logData;
+
+  try {
+    const parsed = JSON.parse(msg);
+    return {
+      ...logData,
+      ...parsed,
+      structured: true
+    };
+  } catch (e) {
+    return {
+      ...logData,
+      structured: false
+    };
+  }
+}
+
+function handleLogData(logger: Logger, logData: any): void {
+  const filtered = {
+    ...logData,
+    pod: undefined,
+    structured: undefined
+  };
+
+  console.log(logger(filtered));
```

Comment:
I would prefer you change the `Logger` interface so that the `Logger` will print to `console.log`. Otherwise, the `Logger` is a `LogFormatter` instead.

Reply:
Okay. Let's rename it to `LogFormatter`.